### PR TITLE
New diary form

### PR DESCRIPTION
**Pull request by neallred**

### CODE REVIEW

File: src/services/__mocks__/diaryApi.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const setItem = () => null
```
Comment (Xesued) [inline src/services/__mocks__/diaryApi.js]: Doesn't block the PR, but I would just set them as `jest.fn()` here, then you can just use them and expect().toBeCalled().

File: src/store/sagas.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import { addDiary } from '../services/diaryApi'
+
+function* saveDiarySaga(action) {
+  const res = addDiary(action.payload)
+
+  // successful save
+  if (typeof res.id === 'string' && res.id.length > 0) {
+    yield put(saveNewDiarySuccess(res))
+    yield put(clearDiaryFields())
```
Comment (Xesued) [inline src/store/sagas.js]: So, every `put` does a dispatch.  Each dispatch can cause a lot of re-rendering or, at the very least, a lot of React reconciling.  

I've found it to be a good practice to try and only dispatch:

1. Actions directly caused by users
2. Actions caused by async actions completing (ex: successful fetch of data from server)

In this instance, I would have the reducer that deals with diary fields listen to the action created by `saveNewDiarySuccess` and clear the fields.

**Review by Xesued — APPROVED** [review]
Looking good!  I would suggest looking into remove the put action and have the reducer itself clear out fields.

File: src/services/__mocks__/diaryApi.js (hunk 1 above)
Comment (neallred) [reply src/services/__mocks__/diaryApi.js]: Nice.

File: src/store/sagas.js (hunk 2 above)
Comment (neallred) [reply src/store/sagas.js]: That's good advice. What else have you found effective at limiting reconciliation/re-rendering? Do you still use re-select or some other form of memoization?

**Pull request merged**